File: backend/appointment-service/src/main/java/org/example/appointmentservice/dto/DoctorDto.java
package org.example.appointmentservice.dto;

import lombok.Data;

import java.math.BigDecimal;

@Data
public class DoctorDto {
    private Integer doctorId;
    private Integer userId;
    private String identityNumber;
    private String fullName;
    private String avatar;
    private String birthday;
    private String gender;
    private String address;
    private String academicDegree;
    private String specialization;
    private String type;
    private BigDecimal consultationFee;
    private Integer departmentId;
    private String departmentName;
    private String createdAt;
}

File: frontend/web/src/role-admin/types/DoctorDto.ts
export type Gender = "MALE" | "FEMALE" | "OTHER";
export type AcademicDegree = "BACHELOR" | "MASTER" | "DOCTOR" | "OTHER";
export type DoctorType = "FULL_TIME" | "PART_TIME" | "OTHER";

export interface DoctorDto {
  doctorId?: number;
  userId: number;
  identityNumber: string;
  fullName: string;
  birthday: string;
  gender: Gender;
  address?: string;
  academicDegree: AcademicDegree;
  specialization: string;
  type: DoctorType;
  departmentId?: number;
  createdAt?: string;
}
